### ISSUE TITLE
Some unittests keep failing

### ISSUE THREAD

**Issue by 130s** [start-jsk/rtmros_hironx#50]
See for example the [travis job in this PReq](https://github.com/start-jsk/rtmros_hironx/pull/49). The job fails due to the failure in running `unittest`. And so far as I've tried number of times, there are `unit test suite` in `hironx_ros_bridge` that fail no matter what. That is a separate issue of `testcases` that need to be fixed. Since this effects my work a lot I'll look into this soon.

**Comment by 130s** [comment]
This is critical; In order to use unittests as a criteria for the regression test, I think it's essential to have testcases that don't fail under good condition. Otherwise we can't tell what causes the tests to fail.

**Comment by k-okada** [comment]
we shold update .travas yaml to check both compile with hrpsys soucecode (for developpers) and hrpsys deb (for released code) code, see
https://github.com/start-jsk/rtmros_common/blob/master/.travis.yml
for example.

still, we'll have problem, I think [1] is the issue. and we'll use retry tag[2] to avoid the problem.
- [ ] [1] [https://code.google.com/p/hrpsys-base/issues/detail?id=192](https://github.com/fkanehiro/hrpsys-base/issues/192)
- [x] [2] https://github.com/start-jsk/rtmros_common/pull/371/files

**Comment by 130s** [comment]
```
we shold update .travas yaml to check both compile with hrpsys soucecode (for developpers) and hrpsys deb (for released code) code, see https://github.com/start-jsk/rtmros_common/blob/master/.travis.yml for example.
```

I see. When I originally opened this ticket I wasn't aware of it; I was only caring the following, but indeed there may be cases when the selection of `hrpsys` source becomes the issue:
- testing with real robot
- pre-release test and build on ROS buildfarm

(I suspect that the majority of hironx users would keep using `.deb` though.)

```
still, we'll have problem, I think [1] is the issue. and we'll use retry tag[2] to avoid the problem.

[1] https://code.google.com/p/hrpsys-base/issues/detail?id=192
[2] https://github.com/start-jsk/rtmros_common/pull/371/files
```

I think because we are aware other issues as follows in addition to those above, we just have to wait for them to be resolved first?

https://github.com/start-jsk/rtmros_hironx/issues/81 (test code is for simulation ? real robot)
https://github.com/start-jsk/rtmros_hironx/issues/82 (test code says OK even if it fail to connect RobotHardware)

**Comment by k-okada** [comment]
> we just have to wait for them to be resolved first?

do you mean that we have wait someone fixed (https://code.google.com/p/hrpsys-base/issues/detail?id=192) ?

**Comment by k-okada** [comment]
currently hironx test code sometime did not pass as follows, maybe somewhere around acceleration test or something.

```
[addJointGroup] group name LARM is already installed
terminate called after throwing an instance of 'CORBA::OBJECT_NOT_EXIST'
testtest_hironx ... ok
testtest_hironx_ik ... ok

[ROSTEST]-----------------------------------------------------------------------

[hironx_ros_bridge.rosunit-test_hironx/test_fullbody_setJointAngles_Clear][passed]
[hironx_ros_bridge.rosunit-test_hironx/test_fullbody_setJointAngles_NoWait][passed]
[hironx_ros_bridge.rosunit-test_hironx/test_fullbody_setJointAngles_Wait][passed]
[hironx_ros_bridge.rosunit-test_hironx/test_fullbody_setJointAngles_minus][passed]
[hironx_ros_bridge.rosunit-test_hironx/test_goInitial][passed]
[hironx_ros_bridge.rosunit-test_hironx/test_rarm_setJointAnglesOfGroup_Override_Acceleration][passed]
[hironx_ros_bridge.rosunit-test_hironx/test_rarm_setJointAnglesOfGroup_minus][passed]
[hironx_ros_bridge.rosunit-test_hironx/test_rarm_setJointAngles_Clear][FAILURE]-
File "/usr/lib/python2.7/unittest/case.py", line 327, in run
    testMethod()
  File "/home/travis/ros/ws_rtmros_hironx/install/share/hironx_ros_bridge/test/test_hironx.py", line 427, in test_rarm_setJointAngles_Clear
    self.check_log_data(data, 6, (5 + clear_time[i]), [(-140+i*40/len(clear_time)),20], -100.0)
  File "/home/travis/ros/ws_rtmros_hironx/install/share/hironx_ros_bridge/test/test_hironx.py", line 148, in check_log_data
    self.assertTrue(flag)
  File "/home/travis/ros/ws_rtmros_hironx/install/share/hironx_ros_bridge/test/test_hironx.py", line 358, in assertTrue
    assert(a)
--------------------------------------------------------------------------------

[hironx_ros_bridge.rosunit-test_hironx/test_rarm_setJointAngles_NoWait][passed]
[hironx_ros_bridge.rosunit-test_hironx/test_rarm_setJointAngles_Wait][passed]
```

**Comment by emijah** [comment]
As a user, I'd just like to mention that for heavy users, who want to modify code, being able to compile from source is crucial.
Thank you for your work, and hope to contribute a bit more in the future.

**Comment by 130s** [comment]
> >    we just have to wait for them to be resolved first?
> 
> do you mean that we have wait someone fixed (https://github.com/fkanehiro/hrpsys-base/issues/192) ?

No, I didn't mean that.

For https://code.google.com/p/hrpsys-base/issues/detail?id=192 in particular, because it looks like happens only once in 10 times or so, it's ignorably trivial isn't it (there are other failure situations that are due to many other random reasons that we don't necessarily investigate)?

**Comment by 130s** [comment]
After a while unit tests are succeeding (eg. within [this travis build 13 tests passed](https://travis-ci.org/start-jsk/rtmros_hironx/builds/25851129)) and they seem to function as their purpose. However, some other issues that are linked from this ticket are not yet resolved. So I still keep this one open in case there might be underlining issues regarding tests.

**Comment by k-okada** [comment]
>  (there are other failure situations that are due to many other random reasons that we don't necessarily investigate)?

I haven't checked the travis log but, test code is sometimes too strict . I'm not sure what's happens exactly but for example if travis is too crowded and there are no enough resources, then the sequencer did not respond as we expected. so sometimes we need to change test condition as
https://github.com/start-jsk/rtmros_hironx/commit/9bbd5412417deb858084c6f565389bafaef6eabb
https://github.com/start-jsk/rtmros_hironx/commit/83886a4d17a16772d7883447e1753363dd4a12a5
ofcourse this increases the potential of passing wrong code.

Anyway we need to look more close on the travis log to identify the cause of failure
